### PR TITLE
CPCM-X

### PR DESCRIPTION
**Pull request by MtoLStoN**
This PR will implement the Extended Conduction-like Polarizable Continuum Solvation Model (CPCM-X) into the xTB suite. 

This should be merged only after the next Bugfix release. 

TODO: Add citation after publication.

### CODE REVIEW

**Comment by MtoLStoN** [comment]
The only thing missing should be the correct citation now after publication (in ``xhelp`` and in the ``Readme``). This should therefore be ready for Review.

**Review by awvwgk — COMMENTED** [review]
Should we make the dependency on CPCM-X optional like the tblite one?

**Comment by MtoLStoN** [comment]
We could probably do this to give the option to compile a lightweight xTB version without the additional dependencies.

**Comment by MtoLStoN** [comment]
That should do the trick.

File: CMakeLists.txt [hunk 1]
```diff
@@ -110,7 +110,7 @@ target_link_libraries(
   ${LAPACK_LIBRARIES}
   "$<$<BOOL:${WITH_OpenMP}>:OpenMP::OpenMP_Fortran>"
   "mctc-lib::mctc-lib"
-  "cpcmx::cpcmx"
+  "$<$<BOOL:${WITH_CPCMX}>:cpcmx::cpcmx>"
   "tblite::tblite"
```
Comment (awvwgk) [inline CMakeLists.txt:114]:
```suggestion
  "$<$<BOOL:${WITH_TBLITE}>:tblite::tblite>"
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -142,7 +142,7 @@ target_link_libraries(
...
```
Comment (awvwgk) [inline CMakeLists.txt:146]:
```suggestion
    "$<$<BOOL:${WITH_TBLITE}>:tblite::tblite>"
```

File: src/solv/cpx.F90 [hunk 3]
```diff
@@ -39,13 +69,16 @@ subroutine setup_cpcmx(self, env, solvent)
         !> TODO: SETUP solute directly from TCosmo instead of reading COSMO file
         Call read_cosmo('xtb.cosmo',self%solute,'NONE',error)
         if (allocated(error)) Call env%error(error%message, source)
+#endif
```
Comment (awvwgk) [inline src/solv/cpx.F90:71]: One thing to do could be adding a message that a routine was called which is not available. This should of course never happen in practice but better safe than sorry ;).

In the tblite part it should be done with something like this:
```suggestion
#else
     call NotImplementedError
#endif
```

Comment (MtoLStoN) [reply src/solv/cpx.F90:73]: I was actually just on it ;).

**Comment by MtoLStoN** [comment]
I added two workflows that test the lightweight xTB build with cmake and meson. Seems to work pretty well.

**Review by Albkat — APPROVED** [review]

**Pull request merged**